src/analysis/sensitivity.jl

```diff
@@ -88,21 +132,43 @@ function pawn(X::T1, y::T2, factor_names::Vector{String}; S::Int64=10)::NamedDim
             p_mean = mean(p_ind)
             p_sdv = std(p_ind)
             p_cv = p_sdv ./ p_mean
-            results[d_i, :] .= (minimum(p_ind), p_mean, median(p_ind), maximum(p_ind), p_sdv, p_cv)
+            results[d_i, :] .= (
+                minimum(p_ind),
+                p_mean,
+                median(p_ind),
+                maximum(p_ind),
+                p_sdv,
+                p_cv
+            )
         end
     end
 
     replace!(results, NaN => 0.0, Inf => 0.0)
 
-    return NamedDimsArray(results; factors=Symbol.(factor_names), Si=[:min, :mean, :median, :max, :std, :cv])
+    col_names = [:min, :mean, :median, :max, :std, :cv]
+    return NamedDimsArray(results; factors=Symbol.(factor_names), Si=col_names)
 end
-function pawn(X::DataFrame, y::AbstractVector; S::Int64=10)::NamedDimsArray
+function pawn(X::DataFrame, y::AbstractVector{<:Real}; S::Int64=10)::NamedDimsArray
     return pawn(Matrix(X), y, names(X); S=S)
 end
-function pawn(X::NamedDimsArray, y::T; S::Int64=10)::NamedDimsArray where {T<:Union{NamedDimsArray,AbstractVector{<:Real}}}
+function pawn(X::NamedDimsArray, y::Union{NamedDimsArray,AbstractVector{<:Real}}; S::Int64=10)::NamedDimsArray
     return pawn(X, y, axiskeys(X, 2); S=S)
 end
-function pawn(rs::ResultSet, y::T; S::Int64=10)::NamedDimsArray where {T<:Union{NamedDimsArray,AbstractVector{<:Real}}}
+function pawn(X::Union{DataFrame,NamedDimsArray}, y::AbstractMatrix{<:Real}; S::Int64=10)::NamedDimsArray
+    N, D = size(y)
+    if N > 1 && D > 1
+        msg::String = string(
+            "The current implementation of PAWN can only assess a single quantity",
+            " of interest at a time."
+        )
+        throw(ArgumentError(msg))
+    end
+
+    # The wrapped call to `vec()` handles cases where matrix-like data type is passed in
+    # (N x 1 or 1 x D) and so ensures a vector is passed along
+    return pawn(X, vec(y); S=S)
+end
+function pawn(rs::ResultSet, y::Union{NamedDimsArray,AbstractVector{<:Real}}; S::Int64=10)::NamedDimsArray
```

Comment:
Function name longer than 92 characters

Reply:
Obviously I'm getting old and going blind.